packages/dataviews/src/view-table.js

```diff
@@ -118,11 +116,12 @@ function HeaderMenu( { dataView, header } ) {
 			align="start"
 			trigger={
 				<Button
-					icon={ sortIcons[ header.column.getIsSorted() ] }
-					iconPosition="right"
-					text={ text }
-					style={ { padding: 0 } }
-				/>
+					size="compact"
+					className="dataviews-table-header-button"
+				>
+					{ text }
+					<span>{ sortArrows[ header.column.getIsSorted() ] }</span>
```

Comment:
Screen readers will read this as "up arrow", or whatever name the system has given to the character, which might not mean much. So we should add an [`aria-hidden="true"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden) here to reduce noise and potential confusion.

We need to address the semantics of this separately anyway (see #56853).

```suggestion
					<span aria-hidden="true">{ sortArrows[ header.column.getIsSorted() ] }</span>
```